### PR TITLE
I500 fix das_t

### PR DESCRIPTION
**Pull request by damhuonglan**
### What does this PR do?
SmartSolo filenames create ordering problem when read into ph5 which the cause of the issue in #500. This PR is to fix the problem:
+ New tool: fix_das_t_order to fixed ordering problem in given ph5. 
**Remember to run: ```pip install -e .``` before using the tool**
+ segdtoph5: reorder das_t for new ph5
+ ph5_validate: give error message for das_t's ordering problem

### Checklist
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [ ] All tests pass.
- [ ] Any new or changed features have are documented.
- [x] Changes have been added to `CHANGELOG.txt` .
- [ ] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by hrotman-pic** [comment]
It looks like this will resolve issue 500.  I'm testing a range of cases before I leave my complete review.

Can I ask for 2 changes (I think they are small):
1) In the help for fix_das_t_order, could you change the statement
"Reorder das_t according to channel_number_i and time."
To instead be:
"Reorder das_t according to channel_number_i and time.  Will update all das_t in an experiment."

2) In segdtoph5, could you change it again so it will reorder Fairfield das_t also?  I am sorry, I updated the issue with a Fairfield example after you made this PR and did not mention that Fairfield could be affected as well as SmartSolo.

File: ph5/utilities/fix_das_t_order.py
Comment (timronan) [inline ph5/utilities/fix_das_t_order.py:95]: Seems to be the core logic of the PR.

**Review by timronan — COMMENTED** [review]
We should merge this PR after we merge #501, so we can roll it back if necessary. The function seems to be working well and the logic seems correct, but I don't have specific Smart-Solo test case like 21-014 to test this on,. @hrotman-pic were you able to test this on a ph5 experiment?

File: ph5/utilities/fix_das_t_order.py
Comment (hrotman-pic) [reply ph5/utilities/fix_das_t_order.py:95]: @timronan Please see my email yesterday about testing & final review of this PR.  Since data extraction is a major test I would like your input.

**Review by timronan — APPROVED** [review]
The update to ph5_validate, additional test cases, and updated change logs are all very useful. Thanks for adding all of this additional content. Because this PR passes all of the unit test cases and the extensive testing that Holly has done, it seems like this PR is ready to be merged. After merging we will use fix_das_t_order to update the PH5 archive.

**Pull request merged**
